gitpod-io/gitpod
# [ws-gc] WS soft deletion improvements

## Description

Makes some improvements on our workspace garbage collection we periodically run:
1. Adds more logging, such as logging the `workspaceMinAgeDays` and `prebuildMinAgeDays` when the job starts, as well as logging every soft- and hard-deleted entry along with its workspace ID and `deletionEligibilityTime`.
2. **Changes the way we decide the `deletionEligibilityTime` of a workspace** from looking at the latest instance's `stoppingTime`, `startedTime` or `creationTime` or workspace's `creationTime` to actually using the current time if we're calling the update as part of a stop, start, create or git status update.
3. Prevents workspaces with a currently running instance from being GC'd

## Related Issue(s)

Fixes ENT-854

## How to test

See [test updates](https://github.com/gitpod-io/gitpod/pull/20271/files#diff-ca6c0a1b67925af3312ecc032b97e88108e3fcd310c9ebfe72c74aeda188cef7)
